spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultPublisherFactory.java

```diff
@@ -53,11 +55,17 @@
 
 	private final ChannelProvider channelProvider;
 
+	private final CredentialsProvider credentialsProvider;
+
 	public DefaultPublisherFactory(String projectId, ExecutorProvider executorProvider,
-			ChannelProvider channelProvider) {
+			CredentialsProvider credentialsProvider) {
 		this.projectId = projectId;
 		this.executorProvider = executorProvider;
-		this.channelProvider = channelProvider;
+		this.channelProvider = TopicAdminSettings.defaultChannelProviderBuilder()
```

Comment:
The idea is that there wouldn't be anything else in `ChannelProvider` that users would want to override. This would also help ensure that the `clientLib` header would be correctly filled.